homeassistant/components/media_player/firetv.py

```diff
@@ -0,0 +1,214 @@
+"""
+homeassistant.components.media_player.firetv
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Provides control over an Amazon Fire TV (/stick) via
+python-firetv, a Python 2.x module with a helper script
+that exposes a HTTP server to fetch state and perform
+actions.
+
+Steps to configure your Amazon Fire TV stick with Home Assistant:
+
+1. Turn on ADB Debugging on your Amazon Fire TV:
+    a. From the main (Launcher) screen, select Settings.
+    b. Select System > Developer Options.
+    c. Select ADB Debugging.
+2. Find Amazon Fire TV device IP:
+    a. From the main (Launcher) screen, select Settings.
+    b. Select System > About > Network.
+3. `pip install firetv[firetv-server]` into a Python 2.x environment
+4. `firetv-server -d <fire tv device IP>:5555`, background the process
+5. Configure Home Assistant as follows:
+
+media_player:
+  platform: firetv
+  # optional: where firetv-server is running (default is 'localhost:5556')
+  host: localhost:5556
+  # optional: device id (default is 'default')
+  device: livingroom-firetv
+  # optional: friendly name (default is 'Amazon Fire TV')
+  name: My Amazon Fire TV
+
+Note that python-firetv has support for multiple Amazon Fire TV devices.
+If you have more than one configured, be sure to specify the device id used.
+Run `firetv-server -h` and/or view the source for complete capabilities.
+
+Possible states are:
+ - off (TV screen is dark)
+ - standby (standard UI is active - not apps)
+ - idle (screen saver is active)
+ - play (video is playing)
+ - pause (video is paused)
+ - disconnected (can't communicate with device)
+"""
+
+import requests
+
+from homeassistant.const import (
+    STATE_PLAYING, STATE_PAUSED, STATE_IDLE, STATE_OFF,
+    STATE_UNKNOWN, STATE_STANDBY)
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice,
+    SUPPORT_PAUSE, SUPPORT_VOLUME_SET,
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_NEXT_TRACK)
+
+SUPPORT_FIRETV = SUPPORT_PAUSE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_PREVIOUS_TRACK | \
+    SUPPORT_NEXT_TRACK | SUPPORT_VOLUME_SET
+
+DOMAIN = 'firetv'
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the firetv platform. """
+    add_devices([
+        FireTVDevice(
```

Comment:
Would it be possible to verify the configuration when loading this platform and only load this platform if the config is working and we can connect to firetv-server?